File: dependencies/src/main/kotlin/com/mthaler/bankapp/service/CustomerRequestServiceImpl.java
package com.mthaler.bankapp.service;

import java.beans.ConstructorProperties;

import com.mthaler.bankapp.dao.CustomerRequestDao;
import com.mthaler.bankapp.domain.CustomerRequestDetails;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

@SuppressWarnings("unused")
public class CustomerRequestServiceImpl implements CustomerRequestService {
	private static Logger logger = LogManager
			.getLogger(CustomerRequestServiceImpl.class);
	private CustomerRequestDetails customerRequestDetails;
	private CustomerRequestDao customerRequestDao;

	@ConstructorProperties({ "customerRequestDetails", "customerRequestDao" })
	public CustomerRequestServiceImpl(
			CustomerRequestDetails customerRequestDetails,
			CustomerRequestDao customerRequestDao) {
		logger.info("Created CustomerRequestServiceImpl instance");
		this.customerRequestDetails = customerRequestDetails;
		this.customerRequestDao = customerRequestDao;
	}

	@Override
	public void submitRequest(String requestType, String requestDescription) {
		// -- create an instance of UserRequestDetails and save it
	}

}

File: context-aware/src/main/kotlin/com/mthaler/bankapp/service/CustomerRequestServiceImpl.kt
package com.mthaler.bankapp.service

import java.beans.ConstructorProperties
import com.mthaler.bankapp.dao.CustomerRequestDao
import com.mthaler.bankapp.domain.CustomerRequestDetails
import org.apache.logging.log4j.LogManager
import org.springframework.context.ApplicationContextAware

class CustomerRequestServiceImpl @ConstructorProperties("customerRequestDetails", "customerRequestDao") constructor(
    customerRequestDetails: CustomerRequestDetails,
    customerRequestDao: CustomerRequestDao
) : CustomerRequestService, ApplicationContextAware {

    private val customerRequestDetails: CustomerRequestDetails
    private val customerRequestDao: CustomerRequestDao

    init {
        logger.info("Created CustomerRequestServiceImpl instance")
        this.customerRequestDetails = customerRequestDetails
        this.customerRequestDao = customerRequestDao
    }

    override fun submitRequest(requestType: String, requestDescription: String) {
        // -- create an instance of UserRequestDetails and save it
    }

    companion object {
        private val logger = LogManager.getLogger(CustomerRequestServiceImpl::class.java)
    }
}